### PR TITLE
fix: ensure correct run-context for 'memcached' instrumentation

### PR DESCRIPTION
**Pull request by trentm**
This ensures that a created 'memcached' span is not the currentSpan
in user code after the call.

Refs: #2430

### Checklist

- [x] Implement code
- [x] Add tests
- [x] Add CHANGELOG.asciidoc entry
- [x] TAV=memcached run

### CODE REVIEW

**Comment by trentm** [comment]
run module tests for memcached

**Comment by trentm** [comment]
run module tests for memcached

**Comment by trentm** [comment]
https://apm-ci.elastic.co/job/apm-agent-nodejs/job/apm-agent-nodejs-mbp/view/change-requests/job/PR-2475/

**Comment by trentm** [comment]
TAV=memcached test run passed: https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-nodejs%2Fapm-agent-nodejs-mbp/detail/PR-2475/4/pipeline

File: lib/instrumentation/modules/memcached.js [hunk 1]
```diff
@@ -34,28 +37,37 @@ module.exports = function (memcached, agent, { version, enabled }) {
...
-              span.end()
-              return origCallback.apply(this, arguments)
-            })
-            // Rewrite the query compiler with the wrapped callback
-            arguments[0] = function queryCompiler () {
-              return query
-            }
-          }
-        }
+    return function wrappedCommand (queryCompiler, _server) {
```
Comment (astorm) [inline lib/instrumentation/modules/memcached.js:40]: nit: it doesn't look like the function ever uses `_server`

**Review by astorm — APPROVED** [review]
👍 Approving.  diff was a little gnarly, but it looks like it does the thing.

File: lib/instrumentation/modules/memcached.js (hunk 1 above)
Comment (trentm) [reply lib/instrumentation/modules/memcached.js:40]: Yah, a "_" prefix on args is my sometimes way of saying "this is the expected position of this argument in the call signature, but I'm not using it". This "_" prefix pattern is this eslint rule (https://github.com/typescript-eslint/typescript-eslint/issues/1510) which I believe is a default in `standard`s set of eslint style rules.

Happy to drop it if you prefer. For trailing args it is purely for self-documenting. For non-trailing args (e.g. https://github.com/elastic/apm-agent-nodejs/blob/3928f8d88d30eb9a012f4b2b5bd9600be2bbc19a/lib/noop-transport.js#L28) it is more helpful/necessary.

**Pull request merged**